nuxeo-notification-stream-core/src/main/java/org/nuxeo/ecm/platform/notification/listener/EventsStreamListener.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * (C) Copyright 2018 Nuxeo SA (http://nuxeo.com/).
+ * This is unpublished proprietary source code of Nuxeo SA. All rights reserved.
+ * Notice of copyright on this source code does not indicate publication.
+ *
+ * Contributors:
+ *     Gildas Lefevre
+ */
+package org.nuxeo.ecm.platform.notification.listener;
+
+import static org.nuxeo.runtime.stream.StreamServiceImpl.DEFAULT_CODEC;
+
+import org.apache.commons.lang3.StringUtils;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.event.Event;
+import org.nuxeo.ecm.core.event.EventBundle;
+import org.nuxeo.ecm.core.event.PostCommitEventListener;
+import org.nuxeo.ecm.core.event.impl.DocumentEventContext;
+import org.nuxeo.ecm.platform.notification.NotificationStreamConfig;
+import org.nuxeo.ecm.platform.notification.message.EventRecord;
+import org.nuxeo.ecm.platform.notification.message.EventRecord.EventRecordBuilder;
+import org.nuxeo.lib.stream.computation.Record;
+import org.nuxeo.lib.stream.log.LogAppender;
+import org.nuxeo.lib.stream.log.LogManager;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.codec.CodecService;
+
+/**
+ * Asynchronous events listener that pushed all the contributed events into a Log to feed the Notification Topology.
+ *
+ * @since XXX
+ */
+public class EventsStreamListener implements PostCommitEventListener {
+
```

Comment:
Using a PostCommitEventListener implies lots of overhead:
- creating a bundle event in an after commit
- schedule an async job
- the job is serialized with the bundle event
- the job is deserialized somewhere the bundle event also
- the event are reconnected, creating session ...
- Then your listener is run to convert event into message

The approach recommended in https://jira.nuxeo.com/browse/NXP-25314
is to follow what is done in StreamAuditEventListener:
- use a sync listener that collect interesting events and deduplicate if needed
- on after commit you write the events to the stream

it is much lighter.